pkg/error/error_test.go

```diff
@@ -112,6 +113,69 @@ func TestGetErrorCount(t *testing.T) {
 	}
 }
 
+func TestValidationErrorEqual(t *testing.T) {
+	baseError := ValidationError{
+		LineNumber: -1,
+		Message:    errors.New("a message"),
+	}
+	wrongLineNumberError := ValidationError{
+		LineNumber: 2,
+		Message:    errors.New("a message"),
+	}
+	differentMessageError := ValidationError{
+		LineNumber: -1,
+		Message:    errors.New("different message"),
+	}
+	differentCountError := ValidationError{
+		LineNumber:       -1,
+		Message:          errors.New("a message"),
+		ConsecutiveCount: 1,
+	}
+	if !baseError.Equal(baseError) {
+		t.Error("failed to detect an error being equal to itself")
+	}
+	if baseError.Equal(wrongLineNumberError) {
+		t.Error("failed to detect a difference in the LineNumber")
+	}
+	if baseError.Equal(differentMessageError) {
+		t.Error("failed to detect a difference in the Message")
+	}
+	if baseError.Equal(differentCountError) {
+		t.Error("failed to detect a difference in the ConsequtiveCount")
+	}
+}
+
+func TestConsolidateErrors(t *testing.T) {
```

Comment:
I would like to have one more tests

About this example

Line 1: error1
Line 2: error2
Line 2: error1
Line 3: error3
Line 3: error1
Line 3: error2
Line 4: error4

I would expect to have:
- error1 detected 3 times starting at line 1
- error2 detected 2 times starting at line 2
- error3
- error4

I'm unsure about what could render the current implementation. And a test like this is important

Reply:
This interleaved pattern is definitely not supported by the current implementation - it would return the list exactly the same as it was put it. As it is currently the implementation is basically a run-length-encoding compression of the error message, nothing more.

Supporting that would require keeping track of multiple different errors - which I'm not confident I could easily do, given my skill level in golang. In my mind that would require first converting the flat list of errors into a map per line, keyed by the error, then scanning that list again to RLE each error type, than merging these results again into a flatter list by sorting by starting line number and error message. I think that would be more than it needs to be, comparing it to the current state running in production.

Reply:
I can understand that.

But I'm afraid, this would be expected.

Said otherwise, I think people will open issues about it.

So it's up to maintainers to choose whether they are OK to merge.

I might be able to help with the implementation in a few days. But I cannot for now, as I only have my phone with me.

Maybe one thing to do is to add the test and use a `t.Skip` to have the test ready for a future refactoring.

Reply:
Did you add such a test for future refactoring?

Reply:
I only came around to doing this just now, but in commit 5a402006d92321a73a19df04714aa07e4f7f5e00 I tried roughly representing the example you gave. I also left a warning for anybody implementing this about the assumption I made.

Reply:
I think we should open an issue about the improvement suggestion that is now in a skipped test

Reply:
Done: #373